### PR TITLE
docs: remove devicemapper

### PR DESCRIPTION
**Pull request by thaJeztah**
- relates to https://github.com/moby/moby/pull/43637

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: docs/deprecated.md [hunk 1]
```diff
@@ -558,12 +558,12 @@ backported), there is no reason to keep maintaining the `overlay` storage driver
 ### Device mapper storage driver
 
 **Deprecated in Release: v18.09**
-**Disabled by default in Release: v23.0**
+**Removed in Release: v24.0**
```
Comment (AkihiroSuda) [inline docs/deprecated.md:561]:
```suggestion
**Removed in Release: v25.0**
```

File: docs/deprecated.md [hunk 2]
```diff
@@ -558,12 +558,12 @@ backported), there is no reason to keep maintaining the `overlay` storage driver
 ### Device mapper storage driver
 
 **Deprecated in Release: v18.09**
-**Disabled by default in Release: v23.0**
```
Comment (AkihiroSuda) [inline docs/deprecated.md:561]: Probably this line can be safely retained

File: docs/deprecated.md (hunk 1 above)
Comment (thaJeztah) [reply docs/deprecated.md:561]: oh! good catch 👍

File: docs/deprecated.md (hunk 2 above)
Comment (thaJeztah) [reply docs/deprecated.md:561]: Works for me; I can add it back

**Comment by thaJeztah** [comment]
@AkihiroSuda updated; PTAL

File: docs/deprecated.md [hunk 3]
```diff
@@ -559,11 +559,12 @@ backported), there is no reason to keep maintaining the `overlay` storage driver
...
 **Deprecated in Release: v18.09**
 **Disabled by default in Release: v23.0**
+**Removed in Release: v25.0**
 
-The `devicemapper` storage driver is deprecated in favor of `overlay2`, and will
-be removed in a future release. Users of the `devicemapper` storage driver are
-recommended to migrate to a different storage driver, such as `overlay2`, which
-is now the default storage driver.
+The `devicemapper` storage driver is deprecated in favor of `overlay2`, and has
+been removed in Docker Engine v24.0. Users of the `devicemapper` storage driver
```
Comment (AkihiroSuda) [inline docs/deprecated.md:565]:
```suggestion
been removed in Docker Engine v25.0. Users of the `devicemapper` storage driver
```

File: docs/deprecated.md [hunk 4]
```diff
@@ -559,11 +559,12 @@ backported), there is no reason to keep maintaining the `overlay` storage driver
...
+must migrate to a different storage driver, such as `overlay2`, before upgrading
+to Docker Engine v24.0.
```
Comment (AkihiroSuda) [inline docs/deprecated.md:567]:
```suggestion
to Docker Engine v25.0.
```

File: docs/deprecated.md (hunk 3 above)
Comment (thaJeztah) [reply docs/deprecated.md:565]: LOL, I really messed up this one 🤦 😂

**Pull request merged**